File: Runtime/Marvin.Runtime.Maintenance/Plugins/DatabaseMaintenance/Wcf/Datamodel.cs
using System.Runtime.Serialization;

namespace Marvin.Runtime.Maintenance.Plugins.DatabaseMaintenance.Wcf
{
    /// <summary>
    /// Conatins the data to the target model.
    /// </summary>
    [DataContract]
    public class DataModel
    {
        /// <summary>
        /// Name of the target model..
        /// </summary>
        [DataMember]
        public string TargetModel { get; set; }

        /// <summary>
        /// Configuration of the database model.
        /// </summary>
        [DataMember]
        public DatabaseConfigModel Config { get; set; }

        /// <summary>
        /// An amount of setups for this model.
        /// </summary>
        [DataMember]
        public SetupModel[] Setups { get; set; }

        /// <summary>
        /// An amount of backups of this model.
        /// </summary>
        [DataMember]
        public BackupModel[] Backups { get; set; }

        /// <summary>
        /// An amount of scripts for this model.
        /// </summary>
        [DataMember]
        public ScriptModel[] Scripts { get; set; }
    }
}

File: src/Marvin.Runtime.Maintenance.Web.UI/src/databases/models/DataModel.ts
import BackupModel from "./BackupModel";
import DatabaseConfigModel from "./DatabaseConfigModel";
import DbMigrationsModel from "./DbMigrationsModel";
import ScriptModel from "./ScriptModel";
import SetupModel from "./SetupModel";

export default class DataModel {
    public TargetModel: string;
    public Config: DatabaseConfigModel;
    public Setups: SetupModel[];
    public Backups: BackupModel[];
    public Scripts: ScriptModel[];
    public AvailableMigrations: DbMigrationsModel[];
    public AppliedMigrations: DbMigrationsModel[];
}
